handlers/handlers.go

```diff
@@ -0,0 +1,55 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/config"
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/mapper"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// ClientError is an interface that can be used to retrieve the status code if a client has errored
+type ClientError interface {
+	Error() string
```

Comment:
It's unnecessary to include Error() in this interface, when you assert for retrieving the statusCode you're only interested in the Code() function.